src/ripple/overlay/impl/Handshake.cpp

```diff
@@ -46,8 +44,8 @@ getFeatureValue(
         return {};
     boost::smatch match;
     boost::regex rx(feature + "=([^;\\s]+)");
-    std::string const value = header->value();
-    if (boost::regex_search(value, match, rx))
+    const std::string allFeatures = std::string{header->value()};
+    if (boost::regex_search(allFeatures, match, rx))
```

Comment:
interesting observation!

boost provides the following typedefs for the `boost::smatch match;` variable. We use it to store the appropriate portion of the input. I couldn't find the correct type for this variable. Ideally, I'd need `match_results<std::string_view::const_iterator>` to be the type of `match`.

`match_results<const char*>` is incorrect, because some inputs might not be null-terminated.
`match_results<std::string::const_iterator>` is type incompatible with `std:string_view value;`

```
typedef match_results<const char*> cmatch;
typedef match_results<std::string::const_iterator> smatch;
#ifndef BOOST_NO_WREGEX
typedef match_results<const wchar_t*> wcmatch;
typedef match_results<std::wstring::const_iterator> wsmatch;
#endif
```

I need to use this overload for the regex_search function.
```
template <class BidiIterator, class Allocator, class charT, class traits>
bool regex_search(BidiIterator first, BidiIterator last, 
                  match_results<BidiIterator, Allocator>& m, 
                  const basic_regex<charT, traits>& e, 
                  match_flag_type flags,
                  BidiIterator base)
```

How can I work around this?